ASP.NET API/WAVC_WebApi/Startup.cs

```diff
@@ -57,58 +64,83 @@ public void ConfigureServices(IServiceCollection services)
-            }).AddJwtBearer(x => {
-                x.RequireHttpsMetadata = false;
-                x.SaveToken = false;
-                x.TokenValidationParameters = new Microsoft.IdentityModel.Tokens.TokenValidationParameters
+                options.DefaultAuthenticateScheme = JwtBearerDefaults.AuthenticationScheme;
+                options.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;
+                options.DefaultScheme = JwtBearerDefaults.AuthenticationScheme;
+            }).AddJwtBearer(builder => {
+                builder.RequireHttpsMetadata = false;
+                builder.SaveToken = false;
+                builder.TokenValidationParameters = new TokenValidationParameters
                 {
                     ValidateIssuerSigningKey = true,
                     IssuerSigningKey = new SymmetricSecurityKey(key),
                     ValidateIssuer = false,
                     ValidateAudience = false,
                     ClockSkew = TimeSpan.Zero
                 };
+                builder.Events = new JwtBearerEvents
+                {
+                    OnMessageReceived = context =>
+                    {
+                        var accessToken = context.Request.Query["access_token"];
+
+                        var path = context.HttpContext.Request.Path;
+                        if (!string.IsNullOrEmpty(accessToken) && (path.StartsWithSegments("/signalR")))
+                        {
+                            context.Token = accessToken;
+                        }
+                        return Task.CompletedTask;
+                    }
+                };
+            });
+            services.AddCors(options =>
+            {
+                options.AddPolicy("MyCors", builder =>
+                {
+                    builder.AllowAnyMethod();
+                    builder.AllowAnyHeader();
+                    builder.AllowCredentials();
+                    builder.WithOrigins("http://localhost:4200");
+                });
             });
+            services.AddSignalR(options => options.EnableDetailedErrors = true);
+            services.AddSingleton<IUserIdProvider, NameUserIdProvider>();
         }
-
-        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
         public void Configure(IApplicationBuilder app, IHostingEnvironment env)
         {
             if (env.IsDevelopment())
             {
                 app.UseDeveloperExceptionPage();
-                app.UseCors(options => {
-                    options.AllowAnyOrigin();
-                    options.AllowAnyMethod();
-                    options.AllowAnyHeader();
-                    });
+                app.UseDatabaseErrorPage();
+
+                app.UseCors("MyCors");
+                
+                //If you want to use SPA runner uncomment code bellow 
+
+                //app.UseSpa(spa =>
```

Comment:
I don't use it because building a whole solution takes to much time. I commented it for your use. When I have my angular server run separately building a aspnet project takes me approx 4-5 seconds instead of 30.

Reply:
But `UseProxyToSpaDevelopmentServer ` hooks up to already running angular project. It doesn't start it, so build time is about the same as without spa at all.

Reply:
Will you apply that? Or now that I explained what that method does maybe you still have any arguments, why not to? If I didn't explain it well enough maybe read about it, but just anwser me :(

Reply:
Just gonna make SPA for myself then